### PR TITLE
Added User.send_message/2

### PR DESCRIPTION
**Pull request by Xh4H**
Added a function to ease the process of sending DMs to users.

poc

### CODE REVIEW

File: lib/coxir/struct/user.ex [hunk 1]
```diff
@@ -102,6 +102,23 @@ defmodule Coxir.Struct.User do
...
+  """
+  @spec send_message(user, String.t) :: map
+
+  def send_message(%{id: id}, content),
+    do: send_message(id, content)
+
+  def send_message(recipient, content) do
+    create_dm(recipient)
+    |> Channel.send_message(content)
+    |> Channel.pretty
```
Comment (satom99) [inline lib/coxir/struct/user.ex]: This is wrong. The function on top returns a [prettified message](https://github.com/satom99/coxir/blob/fca39acc18ac76363a8033cdabea505cfb49b161/lib/coxir/struct/channel.ex#L69) already.

So tldr; remove that line.

File: lib/coxir/struct/user.ex [hunk 2]
```diff
@@ -102,6 +102,23 @@ defmodule Coxir.Struct.User do
...
+  Returns a message object upon success
+  or a map containing error information.
+  """
+  @spec send_message(user, String.t) :: map
+
+  def send_message(%{id: id}, content),
+    do: send_message(id, content)
+
+  def send_message(recipient, content) do
+    create_dm(recipient)
```
Comment (satom99) [inline lib/coxir/struct/user.ex]: Can we turn this into a pipe?

File: lib/coxir/struct/user.ex [hunk 3]
```diff
@@ -102,6 +102,23 @@ defmodule Coxir.Struct.User do
     |> Channel.pretty
   end
 
+  @doc """
+  Sends a DM to a given user.
+
+  Returns a message object upon success
+  or a map containing error information.
```
Comment (satom99) [inline lib/coxir/struct/user.ex]: Can we change this with;

*Refer to `Coxir.Struct.Channel.send_message/2` for more information.*

File: lib/coxir/struct/user.ex [hunk 4]
```diff
@@ -102,6 +102,23 @@ defmodule Coxir.Struct.User do
     |> Channel.pretty
   end
 
+  @doc """
+  Sends a DM to a given user.
```
Comment (satom99) [inline lib/coxir/struct/user.ex]: Can we change DM to *direct message*?

File: lib/coxir/struct/user.ex (hunk 2 above)
Comment (satom99) [reply lib/coxir/struct/user.ex]: Also, at this point we are assuming `create_dm/1` is never going to fail.

Are we safe doing so?

**Review by satom99 — CHANGES_REQUESTED** [review]
A few design changes and a possible issue.

**Comment by Xh4H** [comment]
Changed 💃

**Review by satom99 — APPROVED** [review]

**Comment by satom99** [comment]
Looking good now, thanks!

**Pull request merged**